### PR TITLE
Add vlan tag into the interface xml

### PR DESCRIPTION
**Pull request by yalzhang**
As hostdev interface supports vlan tag, add vlan tag configuration. This is depends on https://github.com/avocado-framework/avocado-vt/pull/2597

Signed-off-by: yalzhang <yalzhang@redhat.com>

### CODE REVIEW

**Comment by yalzhang** [comment]
run all the cases pass except one fail as no enough pci slot
$ avocado run --vt-type libvirt  sriov..ixgbe  
/root/code/avocado/avocado/plugins/run.py:299: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast and keep-tmp. 
  FutureWarning)
JOB ID     : 5006f6ace82a7065372a6b26866d5992aae3abb8
JOB LOG    : /root/avocado/job-results/job-2020-06-17T15.10-5006f6a/job.log
 (01/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.resume_suspend.vf.managed_no.driver.ixgbe: PASS (256.17 s)
 (02/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.resume_suspend.vf.managed_yes.driver.ixgbe: PASS (257.63 s)
 (03/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.resume_suspend.vf_vlan.managed_no.driver.ixgbe: PASS (257.38 s)
 (04/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.resume_suspend.vf_vlan.managed_yes.driver.ixgbe: PASS (258.41 s)
 (05/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.resume_suspend.vf_pool_vlan.vf_list.driver.ixgbe: PASS (257.91 s)
 (06/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.resume_suspend.vf_pool_vlan.pf.managed_yes.driver.ixgbe: PASS (258.11 s)
 (07/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.resume_suspend.macvtap_network.driver.ixgbe: PASS (257.36 s)
 (08/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.resume_suspend.macvtap_network_vlan.driver.ixgbe: PASS (256.45 s)
 (09/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.resume_suspend.macvtap_interface.driver.ixgbe: PASS (255.14 s)
 (10/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.resume_suspend.macvtap_interface_vlan.driver.ixgbe: PASS (255.13 s)
 (11/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.reboot.vf.managed_no.driver.ixgbe: PASS (282.47 s)
 (12/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.reboot.vf.managed_yes.driver.ixgbe: PASS (283.67 s)
 (13/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.reboot.vf_vlan.managed_no.driver.ixgbe: PASS (284.78 s)
 (14/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.reboot.vf_vlan.managed_yes.driver.ixgbe: PASS (282.84 s)
 (15/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.reboot.vf_pool_vlan.vf_list.driver.ixgbe: PASS (284.94 s)
 (16/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.reboot.vf_pool_vlan.pf.managed_yes.driver.ixgbe: PASS (284.87 s)
 (17/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.reboot.macvtap_network.driver.ixgbe: PASS (281.26 s)
 (18/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.reboot.macvtap_network_vlan.driver.ixgbe: PASS (282.38 s)
 (19/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.reboot.macvtap_interface.driver.ixgbe: PASS (281.12 s)
 (20/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.reboot.macvtap_interface_vlan.driver.ixgbe: PASS (280.54 s)
 (21/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.mac_check.vf.managed_no.driver.ixgbe: PASS (173.92 s)
 (22/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.mac_check.vf.managed_yes.driver.ixgbe: PASS (173.80 s)
 (23/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.mac_check.vf_vlan.managed_no.driver.ixgbe: PASS (174.09 s)
 (24/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.mac_check.vf_vlan.managed_yes.driver.ixgbe: PASS (174.43 s)
 (25/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.mac_check.vf_pool_vlan.vf_list.driver.ixgbe: PASS (174.14 s)
 (26/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.mac_check.vf_pool_vlan.pf.managed_yes.driver.ixgbe: PASS (174.15 s)
 (27/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.mac_check.macvtap_network.driver.ixgbe: PASS (172.49 s)
 (28/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.mac_check.macvtap_network_vlan.driver.ixgbe: PASS (172.69 s)
 (29/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.mac_check.macvtap_interface.driver.ixgbe: PASS (170.97 s)
 (30/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.mac_check.macvtap_interface_vlan.driver.ixgbe: PASS (173.02 s)
 (31/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.nop.vf.managed_no.driver.ixgbe: PASS (174.85 s)
 (32/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.nop.vf.managed_yes.driver.ixgbe: PASS (174.10 s)
 (33/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.nop.vf_vlan.managed_no.driver.ixgbe: PASS (174.06 s)
 (34/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.nop.vf_vlan.managed_yes.driver.ixgbe: PASS (172.89 s)
 (35/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.nop.vf_pool.vf_list.driver.ixgbe: PASS (173.77 s)
 (36/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.nop.vf_pool.pf.managed_yes.driver.ixgbe: PASS (174.48 s)
 (37/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.nop.vf_pool.pf.managed_no.driver.ixgbe: PASS (174.11 s)
 (38/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.nop.vf_pool_vlan.vf_list.driver.ixgbe: PASS (174.21 s)
 (39/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.nop.vf_pool_vlan.pf.managed_yes.driver.ixgbe: PASS (173.79 s)
 (40/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.nop.vf_pool_vlan.pf.managed_no.driver.ixgbe: PASS (174.55 s)
 (41/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.nop.macvtap_network.driver.ixgbe: PASS (171.16 s)
 (42/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.nop.macvtap_network_vlan.driver.ixgbe: PASS (170.75 s)
 (43/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.nop.macvtap_interface.driver.ixgbe: PASS (172.91 s)
 (44/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.hotplug.default.nop.macvtap_interface_vlan.driver.ixgbe: PASS (172.81 s)
 (45/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.resume_suspend.vf.managed_no.driver.ixgbe: PASS (244.02 s)
 (46/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.resume_suspend.vf.managed_yes.driver.ixgbe: PASS (242.86 s)
 (47/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.resume_suspend.vf_vlan.managed_no.driver.ixgbe: PASS (245.48 s)
 (48/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.resume_suspend.vf_vlan.managed_yes.driver.ixgbe: PASS (243.72 s)
 (49/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.resume_suspend.vf_pool_vlan.vf_list.driver.ixgbe: PASS (243.90 s)
 (50/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.resume_suspend.vf_pool_vlan.pf.managed_yes.driver.ixgbe: PASS (246.03 s)
 (51/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.resume_suspend.macvtap_network.driver.ixgbe: PASS (245.96 s)
 (52/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.resume_suspend.macvtap_network_vlan.driver.ixgbe: PASS (245.23 s)
 (53/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.resume_suspend.macvtap_interface.driver.ixgbe: PASS (242.94 s)
 (54/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.resume_suspend.macvtap_interface_vlan.driver.ixgbe: PASS (241.35 s)
 (55/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.reboot.vf.managed_no.driver.ixgbe: PASS (268.73 s)
 (56/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.reboot.vf.managed_yes.driver.ixgbe: PASS (269.40 s)
 (57/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.reboot.vf_vlan.managed_no.driver.ixgbe: PASS (271.84 s)
 (58/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.reboot.vf_vlan.managed_yes.driver.ixgbe: PASS (271.25 s)
 (59/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.reboot.vf_pool_vlan.vf_list.driver.ixgbe: PASS (274.02 s)
 (60/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.reboot.vf_pool_vlan.pf.managed_yes.driver.ixgbe: PASS (271.66 s)
 (61/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.reboot.macvtap_network.driver.ixgbe: PASS (269.96 s)
 (62/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.reboot.macvtap_network_vlan.driver.ixgbe: PASS (270.67 s)
 (63/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.reboot.macvtap_interface.driver.ixgbe: PASS (267.17 s)
 (64/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.reboot.macvtap_interface_vlan.driver.ixgbe: PASS (270.24 s)
 (65/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.mac_check.vf.managed_no.driver.ixgbe: PASS (163.12 s)
 (66/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.mac_check.vf.managed_yes.driver.ixgbe: PASS (161.25 s)
 (67/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.mac_check.vf_vlan.managed_no.driver.ixgbe: PASS (160.43 s)
 (68/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.mac_check.vf_vlan.managed_yes.driver.ixgbe: PASS (161.63 s)
 (69/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.mac_check.vf_pool_vlan.vf_list.driver.ixgbe: PASS (161.98 s)
 (70/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.mac_check.vf_pool_vlan.pf.managed_yes.driver.ixgbe: PASS (162.96 s)
 (71/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.mac_check.macvtap_network.driver.ixgbe: PASS (160.29 s)
 (72/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.mac_check.macvtap_network_vlan.driver.ixgbe: PASS (160.60 s)
 (73/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.mac_check.macvtap_interface.driver.ixgbe: PASS (160.17 s)
 (74/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.mac_check.macvtap_interface_vlan.driver.ixgbe: PASS (160.11 s)
 (75/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.nop.vf.managed_no.driver.ixgbe: PASS (161.81 s)
 (76/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.nop.vf.managed_yes.driver.ixgbe: PASS (162.99 s)
 (77/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.nop.vf_vlan.managed_no.driver.ixgbe: PASS (162.13 s)
 (78/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.nop.vf_vlan.managed_yes.driver.ixgbe: PASS (161.68 s)
 (79/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.nop.vf_pool.vf_list.driver.ixgbe: PASS (160.70 s)
 (80/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.nop.vf_pool.pf.managed_yes.driver.ixgbe: PASS (161.70 s)
 (81/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.nop.vf_pool_vlan.vf_list.driver.ixgbe: PASS (162.13 s)
 (82/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.nop.vf_pool_vlan.pf.managed_yes.driver.ixgbe: PASS (161.69 s)
 (83/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.nop.macvtap_network.driver.ixgbe: PASS (159.40 s)
 (84/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.nop.macvtap_network_vlan.driver.ixgbe: PASS (159.91 s)
 (85/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.nop.macvtap_interface.driver.ixgbe: PASS (160.66 s)
 (86/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.nop.macvtap_interface_vlan.driver.ixgbe: PASS (160.11 s)
 (87/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.nfv.one_nic.vf.managed_yes.driver.ixgbe: PASS (161.96 s)
 (88/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.cold_plug.nfv.two_nic.vf.managed_yes.driver.ixgbe: PASS (247.20 s)
 (89/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.scalability_test.multiple_times.nop.vf.managed_yes.driver.ixgbe: PASS (5156.16 s)
 (90/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.guest_with_vf.scalability_test.max_vfs.nop.vf.managed_yes.driver.ixgbe: FAIL: error: Failed to attach device from /var/tmp/xml_utils_temp_oi2g1oir.xml\nerror: internal error: No more available PCI slots\n (363.62 s)
 (91/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.info_check.vf_info.driver.ixgbe: PASS (109.29 s)
 (92/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.info_check.pf_info.driver.ixgbe: PASS (109.49 s)
 (93/98) type_specific.io-github-autotest-libvirt.sriov.normal_test.info_check.vf_order.driver.ixgbe: PASS (108.80 s)
 (94/98) type_specific.io-github-autotest-libvirt.sriov.error_test.invalid_vf_pf.duplicate_vf.guest_with_vf.hotplug.default.nop.vf_pool.vf_list.driver.ixgbe: PASS (72.62 s)
 (95/98) type_specific.io-github-autotest-libvirt.sriov.error_test.invalid_vf_pf.including_pf.guest_with_vf.hotplug.default.nop.vf_pool.vf_list.driver.ixgbe: PASS (72.65 s)
 (96/98) type_specific.io-github-autotest-libvirt.sriov.error_test.invalid_vf_pf.non_existing_pf.guest_with_vf.hotplug.default.nop.vf_pool.pf.managed_yes.driver.ixgbe: PASS (72.68 s)
 (97/98) type_specific.io-github-autotest-libvirt.sriov.error_test.inactive_pool.guest_with_vf.hotplug.default.nop.vf_pool.pf.managed_yes.driver.ixgbe: PASS (72.64 s)
 (98/98) type_specific.io-github-autotest-libvirt.sriov.error_test.domain_save.guest_with_vf.hotplug.default.nop.vf_pool.pf.managed_yes.driver.ixgbe: PASS (174.15 s)
RESULTS    : PASS 97 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 24980.66 s

**Comment by yalzhang** [comment]
no changes for existing cases except enhanced the error check. 
New added 52 cases:
sriov.normal_test.guest_with_vf.hotplug.default.resume_suspend.vf_vlan.managed_no.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.resume_suspend.vf_vlan.managed_yes.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.resume_suspend.vf_pool_vlan.vf_list.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.resume_suspend.vf_pool_vlan.pf.managed_yes.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.resume_suspend.macvtap_network_vlan.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.resume_suspend.macvtap_interface_vlan.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.reboot.vf_vlan.managed_no.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.reboot.vf_vlan.managed_yes.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.reboot.vf_pool_vlan.vf_list.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.reboot.vf_pool_vlan.pf.managed_yes.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.reboot.macvtap_network_vlan.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.reboot.macvtap_interface_vlan.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.mac_check.vf_vlan.managed_no.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.mac_check.vf_vlan.managed_yes.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.mac_check.vf_pool_vlan.vf_list.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.mac_check.vf_pool_vlan.pf.managed_yes.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.mac_check.macvtap_network_vlan.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.mac_check.macvtap_interface_vlan.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.nop.vf_vlan.managed_no.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.nop.vf_vlan.managed_yes.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.nop.vf_pool_vlan.vf_list.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.nop.vf_pool_vlan.pf.managed_yes.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.nop.vf_pool_vlan.pf.managed_no.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.nop.macvtap_network_vlan.driver.ixgbe
sriov.normal_test.guest_with_vf.hotplug.default.nop.macvtap_interface_vlan.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.resume_suspend.vf_vlan.managed_no.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.resume_suspend.vf_vlan.managed_yes.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.resume_suspend.vf_pool_vlan.vf_list.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.resume_suspend.vf_pool_vlan.pf.managed_yes.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.resume_suspend.macvtap_network_vlan.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.resume_suspend.macvtap_interface_vlan.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.reboot.vf_vlan.managed_no.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.reboot.vf_vlan.managed_yes.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.reboot.vf_pool_vlan.vf_list.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.reboot.vf_pool_vlan.pf.managed_yes.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.reboot.macvtap_network_vlan.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.reboot.macvtap_interface_vlan.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.mac_check.vf_vlan.managed_no.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.mac_check.vf_vlan.managed_yes.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.mac_check.vf_pool_vlan.vf_list.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.mac_check.vf_pool_vlan.pf.managed_yes.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.mac_check.macvtap_network_vlan.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.mac_check.macvtap_interface_vlan.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.nop.vf_vlan.managed_no.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.nop.vf_vlan.managed_yes.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.nop.vf_pool_vlan.vf_list.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.nop.vf_pool_vlan.pf.managed_yes.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.nop.macvtap_network_vlan.driver.ixgbe
sriov.normal_test.guest_with_vf.cold_plug.nop.macvtap_interface_vlan.driver.ixgbe
sriov.error_test.vlan_trunk.guest_with_vf.hotplug.default.nop.vf_pool.vf_list.driver.ixgbe
sriov.error_test.vlan_trunk.guest_with_vf.hotplug.default.nop.macvtap_network.driver.ixgbe

File: libvirt/tests/src/sriov/sriov.py [hunk 1]
```diff
@@ -593,11 +599,13 @@ def remove_devices(vmxml, device_type):
     max_vfs_attached = "yes" == params.get("max_vfs_attached", "no")
     inactive_pool = "yes" == params.get("inactive_pool", "no")
     duplicate_vf = "yes" == params.get("duplicate_vf", "no")
-    expected_error = params.get("error_msg", "")
+    expected_error = error_msg = params.get("error_msg", "")
```
Comment (smitterl) [inline libvirt/tests/src/sriov/sriov.py:602]: You don't need this if you use `libvirt.check_result(result, expected_error)`, and you avoid duplication, s. details below.

File: libvirt/tests/src/sriov/sriov.py [hunk 2]
```diff
@@ -710,13 +720,19 @@ def remove_devices(vmxml, device_type):
...
             utils_test.libvirt.check_exit_status(result, expected_error)
+            if error_msg not in result.stderr:
+                test.fail("network define fail with unexpected error!")
         if duplicate_vf:
             vf_list.append(vf_list[0])
             netxml = create_hostdev_network()
             result = virsh.net_define(netxml.xml, ignore_status=True, debug=True)
             utils_test.libvirt.check_exit_status(result, expected_error)
+            if error_msg not in result.stderr:
+                test.fail("network define fail with unexpected error!")
```
Comment (smitterl) [inline libvirt/tests/src/sriov/sriov.py:731]: s. above

File: libvirt/tests/src/sriov/sriov.py [hunk 3]
```diff
@@ -710,13 +720,19 @@ def remove_devices(vmxml, device_type):
...
             result = virsh.net_create(netxml.xml, ignore_status=True, debug=True)
             utils_test.libvirt.check_exit_status(result, expected_error)
+            if error_msg not in result.stderr:
+                test.fail("network create fail with unexpected error!")
```
Comment (smitterl) [inline libvirt/tests/src/sriov/sriov.py:735]: s. above

File: libvirt/tests/src/sriov/sriov.py [hunk 4]
```diff
@@ -710,13 +720,19 @@ def remove_devices(vmxml, device_type):
             netxml = create_hostdev_network()
             result = virsh.net_define(netxml.xml, ignore_status=True, debug=True)
             utils_test.libvirt.check_exit_status(result, expected_error)
+            if error_msg not in result.stderr:
+                test.fail("network define fail with unexpected error!")
```
[the remark this answers is not in the document]
Comment (yalzhang) [reply libvirt/tests/src/sriov/sriov.py]: Thank you for the detailed review and suggestion!
I should read the definition of "utils_test.libvirt.check_result(result, expected_error)" early. Thank you so much!

**Comment by dzhengfy** [comment]
@yalzhang please help fix the conflicts.

**Comment by yalzhang** [comment]
@smitterl  Hi, Thank you for your review, and I have updated the scripts, please help to review again. Thank you!

**Review by chloerh — APPROVED** [review]

**Comment by chloerh** [comment]
@smitterl  Would you like to re-review this pr? If not, I'll get it merged this week.

File: libvirt/tests/src/sriov/sriov.py (hunk 1 above)
Comment (yalzhang) [reply libvirt/tests/src/sriov/sriov.py]: updated， Thank you!

File: libvirt/tests/src/sriov/sriov.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (yalzhang) [reply libvirt/tests/src/sriov/sriov.py]: Updated, Thank you!

File: libvirt/tests/src/sriov/sriov.py (hunk 3 above)
Comment (yalzhang) [reply libvirt/tests/src/sriov/sriov.py]: Updated, Thank you!

File: libvirt/tests/src/sriov/sriov.py [hunk 5]
```diff
@@ -719,10 +727,11 @@ def remove_devices(vmxml, device_type):
             else:
                 new_iface = create_interface()
                 mac_addr = new_iface.mac_address
-            if inactive_pool:
-                result = virsh.attach_device(vm_name, new_iface.xml, flagstr=option,
+            new_iface = create_interface()
```
Comment (kylazhang) [inline libvirt/tests/src/sriov/sriov.py:730]: I saw line 728 also have new_iface, so does it duplicate or some steps need to be removed?

File: libvirt/tests/src/sriov/sriov.py [hunk 6]
```diff
@@ -762,14 +771,14 @@ def remove_devices(vmxml, device_type):
...
             netxml = create_hostdev_network()
             result = virsh.net_define(netxml.xml, ignore_status=True, debug=True)
-            utils_test.libvirt.check_exit_status(result, expected_error)
+            utils_test.libvirt.check_result(result, expected_error)
         if duplicate_vf:
             vf_list.append(vf_list[0])
             netxml = create_hostdev_network()
             result = virsh.net_define(netxml.xml, ignore_status=True, debug=True)
-            utils_test.libvirt.check_exit_status(result, expected_error)
+            utils_test.libvirt.result(result, expected_error)
```
Comment (kylazhang) [inline libvirt/tests/src/sriov/sriov.py:779]: should be check_result?

File: libvirt/tests/src/sriov/sriov.py [hunk 7]
```diff
@@ -785,8 +794,10 @@ def remove_devices(vmxml, device_type):
             logging.debug(vmxml)
             iface_list = create_iface_list(bus_id, nic_num, vf_list)
             for iface in iface_list:
-                process.run("cat %s" % iface.xml, shell=True).stdout_text
-                result = virsh.attach_device(vm_name, iface.xml, flagstr=option,
+                process.run("cat %s" % iface.xml, shell=True)
+                txt = process.run("cat %s" % iface.xml, shell=True).stdout_text
```
Comment (kylazhang) [inline libvirt/tests/src/sriov/sriov.py:798]: there are 2 operation for "cat" so the first one should be removed?

Comment (yalzhang) [reply libvirt/tests/src/sriov/sriov.py:796]: Thank you for observing this! updated.

File: libvirt/tests/src/sriov/sriov.py (hunk 6 above)
Comment (yalzhang) [reply libvirt/tests/src/sriov/sriov.py]: Thank you, updated.

File: libvirt/tests/src/sriov/sriov.py (hunk 5 above)
Comment (yalzhang) [reply libvirt/tests/src/sriov/sriov.py]: delete line 730, Thank you!

**Review by kylazhang — APPROVED** [review]

**Pull request merged**